### PR TITLE
Add `MinBidPrice` in `BaseAuction` 

### LINKED ISSUE #104: Add `MinBidPrice` in `BatchAuction`
## Tasks

- Add `MinBidPrice` field in `BatchAuction`
- Add sanity check in `CreateFixedPriceAuction`, `CreateBatchAuction`, and `PlaceBid`
- Update test cases
- Update docs (`How-To/cli/README.md` and spec docs)

### PR DESCRIPTION
**Pull request by jaybxyz**
## Description

closes: #104 #86 

## Tasks

- [x] Add `MinBidPrice` proto field in `BaseAuction`
- [x] Add getter/setter function in `BaseAuction` interface
- [x] Rename `winning_price` to `matched_price`
- [x] Remove `num_winning_bidders` field 
- [x] Update Msgs
- [x] Update CLI commands
- [x] Add validation checks in `PlaceBid` and `ModifyBid`
- [x] Add/Update test cases (`KeeperTestSuite` -> `MinBidPrice`)
- [x] Update related docs (`x/fundraising/spec/...` `How-To/cli/README.md`

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Appropriate labels applied
- [ ] Targeted PR against correct branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://go.dev/blog/godoc).
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: proto/fundraising/fundraising.proto [hunk 1]
```diff
@@ -222,7 +225,7 @@ message AllowedBidder {
   string bidder = 1;
 
   // max_bid_amount specifies the maximum bid amount that the bidder can bid
-  string max_bid_amount = 2 [
+  string max_bid_amount = 3 [
```
Comment (jeonghoyeo7) [inline proto/fundraising/fundraising.proto]: It seems this must be 2. Or is there other parameter added to `AllowedBidder`?

File: x/fundraising/keeper/bid.go [hunk 2]
```diff
@@ -27,6 +27,10 @@ func (k Keeper) PlaceBid(ctx sdk.Context, msg *types.MsgPlaceBid) (types.Bid, er
 		return types.Bid{}, types.ErrInvalidAuctionStatus
 	}
 
+	if msg.Price.LT(auction.GetMinBidPrice()) {
```
Comment (jeonghoyeo7) [inline x/fundraising/keeper/bid.go:30]: This PR currently has two changes: 1) min bid amount and 2) min bid price. Is 2) min bid price also a target to change?

File: x/fundraising/keeper/bid.go [hunk 3]
```diff
@@ -163,6 +167,10 @@ func (k Keeper) ModifyBid(ctx sdk.Context, msg *types.MsgModifyBid) (types.MsgMo
 		return types.MsgModifyBid{}, types.ErrIncorrectOwner
 	}
 
+	if msg.Price.LT(auction.GetMinBidPrice()) {
```
Comment (jeonghoyeo7) [inline x/fundraising/keeper/bid.go:170]: Same comment as above.

**Review by jeonghoyeo7 — COMMENTED** [review]
I left some minor comments.

File: proto/fundraising/fundraising.proto (hunk 1 above)
Comment (jaybxyz) [reply proto/fundraising/fundraising.proto]: Thanks! Yes, it must be 2.

**Pull request merged**